### ISSUE TITLE
Optional creation of getters/mutations/actions

### ISSUE THREAD

**Issue by intermundos** [eunjae-lee/vuex-dry#1]
Hi,

First of all, thank you for this wonderful tool. Really convenient helper to work with vuex.

Is it possible to disable generation of getters/mutations/actions for Module and use only custom ones?

Thanks in advance!

**Comment by intermundos** [comment]
Hello :) 

One of the use cases is when I dont need all the getters generated since I heavily use $get helper to retrieve data from store. All the getters created are of no particular use. Just showing up in the store in dev tools making it a little messy.

In vuex pathify you can choose between generate getters/actions/mutations for every key or choose specific key. 

I hope I explained well :)

Thanks for your help

**Comment by intermundos** [comment]
I will, only after January 20, after I finish urgent project. Will close for now and keep in touch with you.

Have a nice day there

**Comment by eunjae-lee** [comment]
@intermundos Thanks! Take your time. Before you actually dive in, let's have a short discussion on how to let users configure it.

**Comment by intermundos** [comment]
@eunjae-lee Hello again. 

Sorry, took a little longer than expected :) 

Would love to discuss with you how we can supercharge your tool. Let me know how we can chat about.

**Comment by eunjae-lee** [comment]
Reopening this issue. I guess I closed this issue unintentionally.
Let's have a discussion here to leave all the history.
You said you wanted to disable auto-generation of getters, mutations and actions for specific states.
Could you bring up any example configuration? (maybe something similar to what you liked from vue pathify)
Then we could go on with our discussion over the example.

**Comment by intermundos** [comment]
Sure thing, here will do 👍 

I am using vuex dry in production of a small pwa and it really works great. Now I develop another app with and for me I have found that following "setup" works the best. Even though it is a quick hack and a little thought was put into it :)). I just added syntactic sugar
```js 
import { $get, $commit, $action, get, sync, action } from 'vuex-dry'
```

to the $store and used directly e.g. $store.$get('ui/isLoading'). Very flexible setup and sometimes generation of getters/mutations/actions for particular modules is less needed. 

Perhaps, we can disable generation with flag as in 
```js
config: {
    nonStrictObject: ["user"],
    createGetters: false // maybe this way? or
    createGetters: ["user"] // specify only those you want to auto-generate
  },
```

Something in this direction. What do you think?

**Comment by eunjae-lee** [comment]
Before talking about the config, why did you put `$get` to `$store`?
You can just call `$get("ui/isLoading")` without `$store` in any Vue component.










Djdjdj

**Comment by intermundos** [comment]
I get an error using `$get`. May be I miss something, but this does not work for me. 

`vue.runtime.esm.js?2b0e:619 [Vue warn]: Property or method "$get" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.`

**Comment by eunjae-lee** [comment]
@intermundos are you calling `$get("ui/isLoading")` in your template? If so, `$get` doesn't belong to Vue component, so the error obviously occurs. In your `methods`, you can just use `$get` without any problem. Now I understand you used that hack injecting vuex-dry things into your `$store`. Anyway that's your choice.

Getting back to the main discussion,

```
config: {
    nonStrictObject: ["user"],
    createGetters: false // maybe this way? or
    createGetters: ["user"] // specify only those you want to auto-generate
  },
```

```
config: {
  createGetters: ["user"],
  createMutations: ["user"],
  createActions: ["user"]
}
```

When there's no `createGetters`, vuex-dry, by default, create getters for all states. When there is `createGetters` in the config, vuex-dry create getters for only specified states.

What do you think?

**Comment by intermundos** [comment]
@eunjae-lee I tried `$get` inside methods as well, didn't work. I will recheck once I get home. 

Regarding config part - looks good and easy to use 👍

**Comment by eunjae-lee** [comment]
@intermundos Make sure you configured plugin: https://github.com/eunjae-lee/vuex-dry/blob/master/DOCUMENT.md#install

**Comment by intermundos** [comment]
@eunjae-lee the plugin is configured as described.

**Comment by eunjae-lee** [comment]
@intermundos not sure why it doesn't work. If you file a new issue with more detail, I will check it later.

And please let me know if you have any issue with implementing that config.

**Comment by intermundos** [comment]
@eunjae-lee will dig into it soon and update. 

Sorry for delayed responses, have very limited time lately...:(

**Comment by eunjae-lee** [comment]
@intermundos No worries at all. Take as much time as you want.
